ultraplot/tests/test_geographic.py

```diff
@@ -170,3 +171,411 @@ def test_geoticks_input_handling(recwarn):
     assert len(recwarn) == 0
     # Can parse a string
     ax.format(lonticklen="1em")
+
+
+@pytest.mark.parametrize(
+    ("layout", "lonlabels", "latlabels"),
+    [
+        ([1, 2], "tb", "lr"),
+        ([1, 2], "r", "t"),
+        ([[1, 2, 3], [4, 5, 3]], "t", "lr"),
+    ],
+)
+@pytest.mark.mpl_image_compare
+def test_geoticks_shared(layout, lonlabels, latlabels):
+    fig, ax = uplt.subplots(layout, proj="cyl", share="all")
+    ax.format(
+        latlim=(0, 10),  # smaller rangers are quicker
+        lonlim=(0, 10),
+        lonlines=10,
+        latlines=10,
+        land=True,  # enable land
+        labels=True,  # enable tick labels
+        latticklen=True,  # show ticks
+        lonticklen=True,  # show ticks
+        grid=True,
+        gridminor=False,
+        lonlabels=lonlabels,
+        latlabels=latlabels,
+    )
+    return fig
+
+
+def test_geoticks_shared_non_rectilinear():
+    with pytest.warns(uplt.warnings.UltraPlotWarning):
+        fig, ax = uplt.subplots(ncols=2, proj="aeqd", share="all")
+        ax.format(
+            land=True,
+            labels=True,
+            lonlabels="all",
+            latlabels="all",
+        )
+        fig.canvas.draw()  # draw is necessary to invoke the warning
+    return fig
```

Comment:
This test needs to assert the warning was raise. It does not need to return the fig instance and it probably should close it.

Reply:
You are right about the return although we do have a fixture that closes all figures after each test. I do not follow your first comment though, the `with` block fails if the warning is not given.

Reply:
I removed the closing, in the latest commit. 

I also removed the dead code in the first issue raised. I was first thinking of comparing that the axes are the same but the axes are different objects. Otherwise the tests needs to be expanded. I think checking that that the references of the axes are good enough for now. That is, we now check that the axes being shared matches the same structure as the reference, but we do not check if the properties change with it. This should be covered by other tests however so  I think it is fine as is, but we could add this if that is better.

Reply:
Added a view test in the latest commit